docs/pipelines/cpu-to-carbon.md

```diff
@@ -0,0 +1,359 @@
+4) Determine the power drawn by your CPU by multiplying your scaling factor by the CPU's thermal design power
+5) Perform a unit conversion to convert power in Watts to energy in kwH
+6) Scale the energy estimated for the entire chip to the portion of the chip that is actually in use.
+
+These steps can be executed in IF using just three plugins:
+
+- `Interpolate`
+- `Multiply`
+- `Divide`
+
+
+## Common patterns
+
+The logical flow from CPU utilization to carbon via a power-curve and thermal design power is a common pattern that is likely to be re-used elsewhere.
+
+
+## Constants and coefficients:
+
+| parameter               | description                                                                                                        | value                                              | unit          | source                                                                                                       |
+| ----------------------- | ------------------------------------------------------------------------------------------------------------------ | -------------------------------------------------- |
+| `x`, `y`                | Points on power curve relating CPU utilization to a coefficient used to scale the processor's thermal design power | `x: [0, 10, 50, 100], y: [0.12, 0.32, 0.75, 1.02]` | dimensionless | [Davy, 2021](https://medium.com/teads-engineering/building-an-aws-ec2-carbon-emissions-dataset-3f0fd76c98ac) |
+| `grid-carbon-intensity` | the carbon emitted per unit energy from the electrical grid                                                        | 750                                                | gCO2e/kWh     | approximates global average                                                                                  |
+
+
+## Assumptions and limitations
+
+The following are assumed to be true in this manifest:
+
+- the power curve relating CPU utilization to power is appropriate for the processor being used to run our application
+- the temporal granularity of the observations are sufficient to accurately capture the behaviour of our application
+- the grid carbon intensity is sufficiently accurate for the location where the computational work is done
+
+
+## Components
+
+There is only one component in this example. It represents the entire application. The component pipeline looks as follows:
+
+```yaml
+pipeline:
+  compute:
+    - interpolate
+    - cpu-factor-to-wattage
+    - wattage-times-duration
+    - wattage-to-energy-kwh
+    - calculate-vcpu-ratio
+    - correct-cpu-energy-for-vcpu-ratio
+    - energy-to-carbon
+```
+
+
+## Plugins
+
+### Interpolate
+
+The `interpolate` plugin is used once. The instance is named `interpolate`. It is used to interpolate the curve relating CPU utilization and thermal-design-power factor so that the right value canb be retrived for the observed CPU utilization at each timestep.
```

Comment:
```suggestion
The `interpolate` plugin is used once. The instance is named `interpolate`. It is used to interpolate the curve relating CPU utilization and thermal-design-power factor so that the right value can be retrieved for the observed CPU utilization at each timestep.
```